app/helpers/brevo.py

```diff
@@ -178,17 +193,39 @@ def update_deal_stage(self, data: UpdateDealStageData):
             raise BrevoRequestError(f"Request to Brevo API failed: {e}")
 
     @check_api_key
-    def get_deals_by_pipeline(self, data: GetDealsByPipelineData):
+    def get_all_deals_by_pipeline(self, data: GetAllDealsByPipelineData):
         try:
-            url = f"{self.BASE_URL}crm/deals"
-            params = {
-                "filters[attributes.pipeline]": data.pipeline_id,
-                "limit": data.limit,
-            }
-            response = self._session.get(url, params=params)
-            response.raise_for_status()
-            return response.json()
-        except ApiException as e:
+            all_deals = []
+            offset = 0
+            limit = 100  # Brevo's maximum limit per request (https://developers.brevo.com/docs/api-limits#general-rate-limiting)
+
+            while True:
```

Comment:
Je ne suis pas fan des `while True`, c'est un coup à mal gérer la condition de sortie et se retrouver avec une boucle infinie :/

Comme on connait les rate limit de l'API, je pense qu'il faudrait mettre une condition sur la limite du nombre d'appel et le fait qu'il n'y ai plus de deals à récupérer.

Reply:
Du coup j'ai modifié avec une boucle for avec le maximum de tentative à 100